pickhardtpayments/SyncSimulatedPaymentSession.py

```diff
@@ -259,63 +274,59 @@ def _evaluate_attempts(self, payments):
         total_fees = 0
         paid_fees = 0
         residual_amt = 0
-        number_failed_paths = 0
         expected_sats_to_deliver = 0
         amt = 0
-        print("\nStatistics about {} candidate onions:\n".format(len(payments)))
+        inflight_attempts = []
+        failed_attempts = []
+        print("\nStatistics about {} candidate onions:\n".format(len(attempts)))
+        for attempt in attempts:
```

Comment:
I guess this is motivated by how the code was histroically structured. But wouldn't it make sense to have this attempts list as an `sub_paymet = Payments()` object and later in the loop just add he `sub_payment` to the `payment`?

Reply:
this would allow us that we could later call something like `sub_payment.filter(SettlementStatus.FAILED)` instead of using an additional list